### PR TITLE
m Change code in branch `reviewMaster`

### PR DESCRIPTION
**Pull request by ecotalisman**
Create pullRequest

### CODE REVIEW

File: src/app/ConsoleMenu.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+            menuItem.runApp();
+        }
+    }
+
+    int enterUserInput(String input) {
+        try {
+            return Integer.parseInt(input);
+        } catch (NumberFormatException e) {
+            return 0;
```
Comment (Volodymyr95) [inline src/app/ConsoleMenu.java:32]: why return 0?

Comment (ecotalisman) [reply src/app/ConsoleMenu.java:32]: 1. "If I return any numbers like 0, 4, 5 etc. (excluding 1, 2, 3), the program will operate, but when I input certain letters such as 'sdf' or any other non-numeric characters, the program responds: 'Invalid input! Please enter a valid number...', requiring not just valid numbers but also valid alphanumeric characters."

2. The provided Java code is a method that attempts to parse a string input into an integer. If the string cannot be parsed into an integer (for example, if it contains non-numeric characters), the method catches a NumberFormatException and returns 0.

3. The console output shown demonstrates a prompt to the user to input a number corresponding to an operation: "Encrypt Text File - 1", "Decrypt Text File - 2", or "Exit Console - 3". If the user inputs a non-numeric character or a number outside the given options (such as "123" or "sdf"), the console displays an error message: "Invalid input! Please enter a valid number...". The user is then re-prompted for a valid input.

```
Please enter to number:
Encrypt Text File - 1
Decrypt Text File - 2
Exit Console - 3
123
Invalid input! Please enter a valid number...

Please enter to number:
Encrypt Text File - 1
Decrypt Text File - 2
Exit Console - 3
sdf
Invalid input! Please enter a valid number...

Please enter to number:
Encrypt Text File - 1
Decrypt Text File - 2
Exit Console - 3
```

File: src/app/ConsoleMenu.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+        while (isRunning) {
+            System.out.print("\n" + MENU_INFO);
+            String userInput = scanner.nextLine();
+            MenuOption menuItem = menuItems.getOrDefault(enterUserInput(userInput), new MenuOptionIncorrectItem());
+
+            menuItem.runApp();
+        }
+    }
+
+    int enterUserInput(String input) {
```
Comment (Volodymyr95) [inline src/app/ConsoleMenu.java:28]: why package access modifier?

Comment (ecotalisman) [reply src/app/ConsoleMenu.java:28]: The use of the no modifier (package-private) access level for **enterUserInput** allows it to be used within the app package. This enables the method to be overridden within the same package if there is a need to create some additional menu for this package. This approach provides a middle ground between public accessibility and more restrictive levels of access control. It allows classes that are closely related to each other (i.e., within the same package) to share certain methods and fields, while still preventing unrestricted access from outside classes.

This information about packages was obtained from the section about access control in Java, as per the official Oracle Java documentation: [Access Levels](https://docs.oracle.com/javase/tutorial/java/javaOO/accesscontrol.html).

**Review by Volodymyr95 — APPROVED** [review]

**Pull request merged**